platform/core/src/services/ToolBarService/ToolbarService.ts

```diff
@@ -135,48 +152,48 @@ export default class ToolbarService extends PubSubService {
       return;
     }
     const commandsManager = this._commandsManager;
-    const { groupId, itemId, commands, type } = interaction;
-    let { interactionType } = interaction;
+    const { commands, type, id, groupId } = interaction;
 
-    // if not interaction type, assume the type can be used
-    if (!interactionType) {
-      interactionType = type;
+    const itemId = interaction.itemId || id;
+
+    const buttonProps = this.getButtonProps(itemId as string);
+
+    if (!buttonProps) {
+      throw new Error(`Button not found for id: ${itemId}`);
     }
 
+    // if not interaction type, assume the type can be used
+    const { interactionType = type } = interaction;
+
+    const prevPrimaryToolId = this.state.primaryToolId;
+
     switch (interactionType) {
-      case 'action': {
+      case ButtonInteractionType.ACTION: {
```

Comment:
Could we get rid of button interaction type, and require toggle buttons to implement a callback that gets the current button state/class set?  That way changing the viewport or camera or whatever has listeners can trigger toggle button updates as required, and if the action fails to trigger, it won't get set as being toggled.